internal/alb/lb/loadbalancer.go

```diff
@@ -255,6 +259,34 @@ func (controller *defaultController) reconcileLBInstance(ctx context.Context, in
 	return nil
 }
 
+func (controller *defaultController) reconcileShield(ctx context.Context, lbArn string, enableProtection bool) error {
+	available, err := controller.cloud.ShieldAvailable(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to get shield advanced subscription state %v", err)
+	}
+	if enableProtection && !available {
+		return fmt.Errorf("unable to create shield advanced protection for loadBalancer %v, shield advanced subscription is not active", lbArn)
+	}
+
+	protection, _ := controller.cloud.GetProtection(ctx, aws.String(lbArn))
```

Comment:
`isProtectionEnabled`? Also if this errored out you will enabled protection which might not be wanted in a transient error for get?

Reply:
protection is either existing protection object or nil if there are currently no protection for the given load balancer resource. Added check for the not-found path, and error checks for failures.